linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -737,6 +737,11 @@ private final class Analyzer(config: CommonPhaseConfig,
           (isScalaClass || isJSClass || isNativeJSClass)) {
         isInstantiated = true
 
+        for (className <- data.referencedFieldClasses) {
```

Comment:
Hum, I just realized that this is going to miss references to the types of *static* fields if the class is accessed but not instantiated. At least, here, we should only process instance fields. Static fields should be dealt in a different way.

Now, actually static fields might not need any specific treatment, for a side-reason: they are reached iff they are read or written, but when they are read or written, we'll also reach their type because it is the `tpe` of the `SelectStatic`. It's a bit of a roundabout way to look at things, though. :-s If we rely on that we should document it in a comment somewhere.